src/utils/PathUtils.ts

```diff
@@ -106,13 +107,98 @@ function convertSoftIndexes(element: PathPart, pathMap: Map<string, number>) {
   }
 }
 
+/**
+ *
+ * @param {PathPart} element - A single element in a rules path
+ * @param {Map<string, number} pathMap - A map containing an element's name as the key and that element's most recently used index as the value
+ * @param {Map<string, number} maxPathMap - A map containing an element's name as the key and that element's maximum index as the value
+ */
+function convertSoftIndexesStrict(
```

Comment:
You're just following the precedent set by `convertSoftIndexes`, but isn't the plural of `index` `indices`?